src/include/platform/internal/DeviceNetworkInfo.h

```diff
@@ -0,0 +1,74 @@
+/*
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *    Copyright (c) 2018 Nest Labs, Inc.
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#ifndef DEVICE_NETWORK_INFO_H
+#define DEVICE_NETWORK_INFO_H
+
+#include <stdint.h>
+
+namespace chip {
+namespace DeviceLayer {
+namespace Internal {
+
+/**
+ * Ids for well-known network provision types.
+ */
+enum
+{
+    kThreadNetworkId            = 1,
+    kWiFiStationNetworkId       = 2,
+    kMaxThreadNetworkNameLength = 16,
+    kThreadExtendedPANIdLength  = 8,
+    kThreadMeshPrefixLength     = 8,
+    kThreadNetworkKeyLength     = 16,
+    kThreadPSKcLength           = 16,
+    kThreadChannel_NotSpecified = UINT8_MAX,
```

Comment:
I believe this may be an unusual enum usage. Enum seems to be an enumeration of distinct values. Here tehe values are not distinct (lengths may have similar values) and not as related.

I believe constexpr actual constants would be better than enums.

Reply:
I'm already using constexprs in the latest commit.

Reply:
Done